src/__tests__/Store.js

```diff
@@ -740,6 +745,88 @@ describe('requests', () => {
             expect(animalStore.isLoading).toBe(false);
         });
     });
+
+    test('handles out-of-order responses', async () => {
+        const animalStore = new AnimalStore();
+
+        // slow fetch, represents fetching an entire store
+        mock.onAny().replyOnce(() => {
+            return delay(1000).then(() => {
+                return [200, animalsData];
+            });
+        });
+
+        // Request 2 - arrives immediately
+        mock.onAny().replyOnce(() => {
+            return [200, animalsDataIdOne];
+        });
+
+        // Fire both requests roughly at the same time
+        const p1 = animalStore.fetch();
+        const p2 = animalStore.fetch({ params: { id: 1 } });
+
+        await Promise.all([p1, p2]);
+
+        // p2 is called last, so will be set
+        expect(animalStore.models.length).toBe(1);
```

Comment:
I would also check `__pendingRequestCount` here: just 1 line of code more is needed